openpgp/v2/keys.go

```diff
@@ -145,10 +180,17 @@ func (e *Entity) EncryptionKey(now time.Time, config *packet.Config) (Key, bool)
 			PublicKey:            e.PrimaryKey,
 			PrivateKey:           e.PrivateKey,
 			SelfSignature:        primarySelfSignature,
-		}, true
+		}, nil
+	}
+
+	if latestSelectionError == nil {
+		latestSelectionError = &errors.ErrEncryptionKeySelection{
+			PrimaryKeyId:    e.PrimaryKey.KeyIdString(),
+			EncSelectionErr: goerrors.New("no key with encryption flags found"),
```

Comment:
It's a minor nitpick but `isValidEncryptionKey` returns false not only if the encryption flags aren't set but also if the algorithm isn't encryption-capable. Normally those shouldn't have encryption flags either but you never know..
```suggestion
			EncSelectionErr: goerrors.New("no encryption-capable key found"),
```
Alternatively, if you want to have a more specific error, you could split out the check for `algo.CanEncrypt()` from `isValidEncryptionKey` into the loop above, and have specific errors for both cases.

Reply:
Uh great catch. I think it is fine with the updated error message. I also added the reason: `no encryption-capable key found (no key flags or invalid algorithm)`,